apps/sms/test/marionette/lib/messages.js

```diff
@@ -350,6 +361,22 @@
             event.initEvent('action', true, true);
             header.dispatchEvent(event);
           });
+        },
+
+        setStorage: function(threads) {
+          client.executeScript(function(threads) {
+            var threadMap = new Map(threads.map(function(thread) {
+              return [thread.id, thread];
+            }));
+
+            window.wrappedJSObject.TestStorages.setStorage('messagesDB', {
+              threads: threadMap,
+              recipientToThreadId: new Map(),
+              messageIdUniqueCounter: 100
+            });
+
+            window.wrappedJSObject.TestStorages.setStorageReady('messagesDB');
```

Comment:
Can't really think of anything better than this, but the idea is the following: 

1. There is some fake storage that mocks use internally;
2. This storage should be filled from test code to simulate various scenarios;
3. Mocks are injected when app is started (via ```contentScript.inject``` plugin);
4. Here is the main problem that influenced solution - we have to fill storage with fake data before application requests it, to solve this I have the following flow - ```app.getData() -> mock.getData() -> TestStorages.getStorage('key') -> storage.getData()```. Specific test/suite decides when ```TestStorages.getStorage('key')``` resolves and application gets the data, so that test creates fake data, populates storage and then marks that storage as ```ready``` and app request passes through.

Alternative would be dedicated files with fake data for various scenarios that is injected before test/suite, but test won't have chance to work with that data, it can only choose what workload to load before app is run.

Do you have any better idea in mind?